### PR TITLE
CDAP-15642 Pubsub integration tests coverage

### PR DESCRIPTION
**Pull request by vladhlinskiy**
JIRA: https://issues.cask.co/browse/CDAP-15642

In the scope of this PR:

Created PubSub integration tests

### CODE REVIEW

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+  }
+
+  @Override
+  public void innerTearDown() {
+    for (ProjectTopicName topic : topicsToDelete) {
+      try {
+        topicAdmin.deleteTopic(topic);
+        LOG.info("Deleted topic {}", topic.toString());
+      } catch (ApiException ex) {
+        LOG.warn("Failed to delete topic {}", topic.toString());
```
Comment (mlozbin-cybervisiontech) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java]: I think you should also log error stacktrace here

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+      }
+    }
+    for (ProjectSubscriptionName subscription : subscriptionsToDelete) {
+      try {
+        subscriptionAdmin.deleteSubscription(subscription);
+        LOG.info("Deleted subscription {}", subscription.toString());
+      } catch (ApiException ex) {
+        LOG.warn("Failed to delete subscription {}", subscription.toString());
```
Comment (mlozbin-cybervisiontech) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java]: I think you should also log error stacktrace here

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+    public void receiveMessage(PubsubMessage pubsubMessage, AckReplyConsumer ackReplyConsumer) {
+      ackReplyConsumer.ack();
+      String msgJson = pubsubMessage.getData().toStringUtf8();
+      CDAPPubSubMessage msg = gson.fromJson(msgJson, CDAPPubSubMessage.class);
+      String messageString = msg.getMessageString();
+      putMessage(messageString);
+      LOG.info("Retrieved PubSub message: {}", messageString);
+    }
+
+    synchronized List<String> getMessages() {
```
Comment (mlozbin-cybervisiontech) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java]: private?

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+      return new ArrayList<>(messages);
+    }
+
+    synchronized int getMessageCount() {
+      return messages.size();
+    }
+
+    private synchronized void putMessage(String message) {
+      messages.add(message);
```
Comment (mlozbin-cybervisiontech) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java]: you can simply use concurrent version of list

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+    }
+
+    void assertMessages(long timeout, TimeUnit unit, String... messages) throws Exception {
```
Comment (mlozbin-cybervisiontech) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java]: assertRetrievedMessagesContain

**Review by mlozbin-cybervisiontech — CHANGES_REQUESTED** [review]

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java (hunk 1 above)
Comment (vladhlinskiy) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java]: Thanks, changed.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java (hunk 2 above)
Comment (vladhlinskiy) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java]: Thanks, changed.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java (hunk 3 above)
Comment (vladhlinskiy) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java]: Removed completely.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java (hunk 4 above)
Comment (vladhlinskiy) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java]: Replaced on CopyOnWriteArrayList.

**Pull request merged**